src/com/amazon/ion/impl/IonReaderTextRawTokensX.java

```diff
@@ -1205,6 +1206,9 @@ protected void skip_over_sexp() throws IOException
     }
     private void skip_over_container(int terminator) throws IOException
     {
+        final int CONTAINER_STACK_INITIAL_CAPACITY = 16;
+        final ArrayList<Integer> terminatorStack = new ArrayList<>(CONTAINER_STACK_INITIAL_CAPACITY);
```

Comment:
Let's move this to the class level and just clear it upon entering this method, to avoid allocating a new one every time.

Reply:
I thought about that, but does that risk introducing thread-safety issues?

Reply:
Oh wait, we construct a new reader instance for each input source, so it wouldn't introduce any thread safety issues that aren't already present.